core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -239,4 +259,37 @@ export class NavigationComponent implements OnInit {
   public hasOperators(): boolean {
     return this.workflowActionService.getTexeraGraph().getAllOperators().length > 0;
   }
+
+  public onClickSaveWorkflow(): void {
+    if (!this.userService.isLogin()) {
+      alert('please login');
+    } else {
+      this.isSaving = true;
+      const cachedWorkflowStr = this.cachedWorkflowService.getCachedWorkflow();
+      if (cachedWorkflowStr != null) {
+        this.workflowPersistService.saveWorkflow(cachedWorkflowStr, this.currentWorkflowName).subscribe(
+          (workflow: Workflow) => {
+            localStorage.removeItem('wfId');
+            localStorage.setItem('wfId', JSON.stringify(workflow?.wfId));
+          }).add(() => {
+          this.isSaving = false;
+        });
+      } else {
+        alert('No workflow found in cache.');
+      }
+
+    }
+  }
+
+  onWorkflowNameChange() {
+    localStorage.setItem('workflowName', this.currentWorkflowName);
+  }
+
+  onClickCreateNewWorkflow() {
+    localStorage.removeItem('workflow');
+    localStorage.removeItem('wfId');
+    localStorage.removeItem('workflowName');
+    this.currentWorkflowName = localStorage.getItem('workflowName') ?? 'Untitled Workflow';
+    this.cachedWorkflowService.loadWorkflow();
```

Comment:
clean `undo` and `redo` as well.

Reply:
this part needs to be done after changing the usage of localStorage. Will do it separately.